device/iot-device-client/devdoc/requirement_docs/com/microsoft/azure/iothub/diagnosticpropertydata_requirements.md

```diff
@@ -0,0 +1,75 @@
+# DiagnosticPropertyData Requirements
+
+## Overview
+
+A message of diagnostic data
+
+## References
+
+## Exposed API
+
+```java
+public class DiagnosticPropertyData
+{
+    public DiagnosticPropertyData(String diagnosticId, String diagnosticCreationTimeUtc);
+    public String getDiagnosticId();
+    public void setDiagnosticId(String diagnosticId);
+    public String getDiagnosticCreationTimeUtc();
+    public void setDiagnosticCreationTimeUtc(String diagnosticCreationTimeUtc);
+    public String getCorrelationContext();
+}
+```
+## Validation
+**SRS_DIAGNOSTICPROPERTYDATA_02_001: [**A valid `diagnosticId` shall not be null or empty.**]**  
+**SRS_DIAGNOSTICPROPERTYDATA_02_002: [**A valid `diagnosticCreationTimeUtc` shall not be null or empty.**]**  
+
+### DiagnosticPropertyData
+
+```java
+public DiagnosticPropertyData(String diagnosticId, String diagnosticCreationTimeUtc);
+```
+
+**SRS_DIAGNOSTICPROPERTYDATA_01_001: [**The constructor shall save the message body.**]**
+
+**SRS_DIAGNOSTICPROPERTYDATA_01_002: [**If the diagnosticId or diagnosticCreationTimeUtc is null, the constructor shall throw an IllegalArgumentException.**]**
+
+
+### getDiagnosticId
+
+```java
+public String getDiagnosticId();
+```
+
+**SRS_DIAGNOSTICPROPERTYDATA_01_003: [**The function shall return the value of diagnostic id.**]**
+
+### setDiagnosticId
+
+```java
+public void setDiagnosticId(String diagnosticId);
```

Comment:
Why do you need this setter isn't it already provided in constructor ? can diagnostic ID change over time ?